src/main/java/seedu/rc4hdb/logic/commands/modelcommands/ListCommand.java

```diff
@@ -15,10 +18,38 @@ public class ListCommand extends ModelCommand {
 
     public static final String MESSAGE_SUCCESS = "Listed all persons";
 
+    public static final String SHOW_ONLY_SPECIFIED = "Listed all persons. Only specified fields are shown.";
+
+    private final List<String> fieldList;
+
+    /**
+     * Constructor for a ListCommand instance.
+     */
+    public ListCommand() {
+        // Set the required list of fields to the global field list
+        this.fieldList = Fields.FIELDS;
+    }
+
+    /**
+     * Constructor for a ListCommand instance.
+     * @param fieldList The fields to be hidden when listing the data
+     */
+    public ListCommand(List<String> fieldList) {
+        requireNonNull(fieldList);
+        this.fieldList = fieldList;
+    }
+
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
         model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
-        return new CommandResult(MESSAGE_SUCCESS);
+        model.setObservableFields(fieldList);
+
+        // Determine which ListCommand constructor was invoked
+        if (fieldList == Fields.FIELDS) {
+            return new CommandResult(MESSAGE_SUCCESS);
+        } else {
+            return new CommandResult(SHOW_ONLY_SPECIFIED);
+        }
```

Comment:
Most of the logic required for implementing `HideCommand` and `ShowCommand` has already been implemented in this commit, so it wouldn't be too bad to do it as such.

Reply:
Hi Jordan, thank you for your suggestion. I agree that allowing for the specified columns to persist between commands would be more useful as a feature. Is it alright if I implement the suggested increment as a feature in v1.3? Currently, the command test PR needs to be urgently settled, so I think it would be good if I lent Alvin a hand. 😄

Reply:
I've created an issue for this #84.